File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Java/fc8417b105c84cf8d2d99d8b0554ab73.java
import java.util.*;

public class fc8417b105c84cf8d2d99d8b0554ab73_nc {
    public static void main(String[] args) {
        int ans = Integer.MAX_VALUE;
        int n, k;
        Map<Integer, Boolean> t = new HashMap<>();
        List<Integer>[] a = new ArrayList[200099];
        for (int i = 0; i < 200099; i++) {
            a[i] = new ArrayList<>();
        }
        Scanner scanner = new Scanner(System.in);
        n = scanner.nextInt();
        k = scanner.nextInt();
        for (int i = 0, temp, cnt = 0; i < n; i++, cnt = 0) {
            temp = scanner.nextInt();
            while (temp != 0) {
                a[temp].add(cnt++);
                t.put(temp, true);
                temp /= 2;
            }
        }
        for (Map.Entry<Integer, Boolean> entry : t.entrySet()) {
            int i = entry.getKey();
            if (a[i].size() < k) {
                continue;
            }
            Collections.sort(a[i]);
            int sum = 0;
            for (int j = 0; j < k; j++) {
                sum += a[i].get(j);
            }
            ans = Math.min(ans, sum);
        }
        System.out.println(ans);
    }
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Python/fc8417b105c84cf8d2d99d8b0554ab73.py
import sys
from typing import List, Tuple
from collections import defaultdict

def minimum_sum(n: int, k: int, nums: List[int]) -> int:
    ans = sys.maxsize
    t = defaultdict(bool)
    a = defaultdict(list)
    
    for i in range(n):
        temp = nums[i]
        cnt = 0
        while temp:
            a[temp].append(cnt)
            t[temp] = True
            temp //= 2
            cnt += 1
    
    for i in t:
        if len(a[i]) < k:
            continue
        a[i].sort()
        ans = min(ans, sum(a[i][:k]))
    
    return ans

if __name__ == "__main__":
    n, k = map(int, input().split())
    nums = list(map(int, input().split()))
    result = minimum_sum(n, k, nums)
    print(result)
